dataflow/src/main/java/org/checkerframework/dataflow/cfg/DOTCFGVisualizer.java

```diff
@@ -180,8 +177,10 @@ protected String dotOutputFileName(UnderlyingAST ast) {
         } else {
             throw new BugInCF("Unexpected AST kind: " + ast.getKind() + " value: " + ast);
         }
-        outFile.append("-");
-        outFile.append(checkerName);
+        if (!checkerName.isEmpty()) {
```

Comment:
Here we don't need to check if `checkerName != null` since all the usages of it make sure that `checkerName` will never be `null`.

In playground, `checkerName` will be set as `""`:
https://github.com/typetools/checker-framework/blob/master/dataflow/src/main/java/org/checkerframework/dataflow/cfg/JavaSource2CFGDOT.java#L154

Reply:
Can you improve error handling?
I would add a null check here and make that field nullable. Maybe the configuration map doesn't contain that key and that is ok.
In contrast, field `outDir` must be non-null and that configuration always must be provided. So add a check for that value to be non-null in the constructor.